scanpy/tests/test_clustering.py

```diff
@@ -20,6 +21,52 @@ def test_leiden_basic(adata_neighbors):
     sc.tl.leiden(adata_neighbors)
 
 
+def test_clustering_subset(adata_neighbors):
+    methods = (
+        (sc.tl.louvain, 'louvain'),
+        (sc.tl.leiden, 'leiden'),
+    )
+    for clustering, key in methods:
```

Comment:
Instead of this loop, you should use `pytest.mark.parametrize`.

This will make two tests out of one, which is nicer if only one of the two breaks.

```py
@pytest.mark.parametrize('clustering,key', [
    (sc.tl.louvain, 'louvain'),
    (sc.tl.leiden, 'leiden'),
])
def test_clustering_subset(adata_neighbors, clustering, key):
    clustering(adata_neighbors, key_added=key)
    print(adata_neighbors)
    ...
```

Reply:
Cool! I didn't know it. It will be in the next commit.